lib/clerk/rack_middleware_v2.rb

```diff
@@ -163,11 +163,14 @@ def call(env)
       end
 
       token = verify_token(cookie_token)
+      return signed_out(env) unless token
 
-      if token && token["iat"] && client_uat && Integer(client_uat) <= token["iat"]
+      if token["iat"] && client_uat && Integer(client_uat) <= token["iat"]
         return signed_in(env, token, cookie_token)
+      else
+        return unknown(interstitial: true)
       end
-
+    rescue JWT::ExpiredSignature, JWT::InvalidIatError => e
```

Comment:
Why does this wrap the whole function? Can we limit it to the places where we expect this error to be raised?

Reply:
see: https://github.com/clerkinc/clerk-sdk-ruby/pull/30/files#r1126216650

Reply:
Let's only wrap the calls where this error may be raised, as opposed to the whole function. Otherwise the code becomes harder to read since you have to take into account all the potential places where this error can be raised.

Reply:
code updated. Could you take another look?